### ISSUE TITLE
configurable email notifications

### ISSUE THREAD

**Issue by grahamlopez** [adam-mcdaniel/rusty-ci#57]

**Comment by adam-mcdaniel** [comment]
PR #60 adds some functionality for this.

For now, rusty-ci can
- Read mail info from an external file to prevent sensitive authentication data from being added to the repo
- Email all fails / successes to `extra-recipients`
- Email all interested users (anyone who has a commit in the PR)
- Send updates from a `from-address` given an SMTP host, port, and password

**Comment by grahamlopez** [comment]
- we usually don't require a password for smtp authentication
- we want to be able to opt in/out of emails for fails and/or successes

**Comment by adam-mcdaniel** [comment]
The email yaml file is optional, you can build it with the `-m` or `--mail` flag. If you don't use the mail yaml file, mail notifications wont be included in the output configuration files

**Comment by grahamlopez** [comment]
I mean that we want to be able to choose only failures, or both failures and successes, or only successes, or nothing

**Comment by adam-mcdaniel** [comment]
I'm adding that functionality now.

It would look something like this


```yaml
extra-recipients:
  adam.mcdaniel17@gmail.com:
    - failure
    - success
  other@example.org:
    - all
```

**Comment by adam-mcdaniel** [comment]
On second thought, it would be easier to implement like so

```yaml
extra-recipients:
  failure:
    - admin1@github.com
  success:
    - adam.mcdaniel17@gmail.com
  all:
    - other@example.org
```

**Comment by adam-mcdaniel** [comment]
You can now configure `extra-recipients` to receive failures, successes, and all info.

**Comment by adam-mcdaniel** [comment]
Fixed with #62
